### ISSUE TITLE
Make sequence a newtype

### ISSUE THREAD

**Issue by Kixunil** [rust-bitcoin/rust-bitcoin#1082]
Making sequence a simple newtype could avoid accidentally mixing it up with other things. It looks like there's such [a bug in `miniscript`](https://github.com/rust-bitcoin/rust-miniscript/pull/408#discussion_r911272745) but regardless of whether that one is a bug newtypes are great in security-critical code like Bitcoin.

Additionally, we can have a bunch of handy methods and constants such as `is_final()`, `is_rbf()`...

Proposed declaration:

```rust
#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
pub struct Sequence(u32); // perhaps the field could be `pub`?

// impl Display for Sequence
// Any interesting (math) operators? Seems too much to me but maybe someone has a use case
```

The type is a bit weird because `u32::MAX` means infinity but it's probably OK or `Ord`/`Eq` and we can be extra careful for other operations.

**Comment by dr-orlovsky** [comment]
@Kixunil if you are interested, that's how I did `SeqNo` type some time ago: https://docs.rs/descriptors/0.8.0/src/descriptors/locks.rs.html#96

**Comment by tcharding** [comment]
Warning to anyone reading the good-first-isue, this is very likely _not_ a trivial issue to do because nSequence is fraught with corner cases and this involves API design, which will mean the PR that solves this issue will likely require a lot of review before it gets merged. Its a good-first-issue in that you will really have to learn about timelocks in bitcoin to do this one though :) I have this queued up in my todo list after #994 gets done but any one is free to jump in.

**Comment by nlanson** [comment]
@tcharding If you don't mind I can get this done within the next day or so.

**Comment by Kixunil** [comment]
@tcharding I think it largely depends on how well one knows Rust. This is the kind of newtype that's very easy. Review will resolve any issues and if my review will help someone learn Rust&Bitcoin better I'll happily do it. :)

**Comment by tcharding** [comment]
Go for it @nlanson, don't worry about merge order in relation to rebasing, I was waiting for #994 because I thought many of the same concepts (and problems) would come up while do nLockTime (its the less complicated of the two Bitcoin time locks). But by all means go for it, there is only an issue with `Ord` to be resolved in #994 (hopefully).
